### DISCUSSION TITLE
Really slow on chrome 85 and up

### DISCUSSION THREAD

**Discussion by mgibas** [tailwindlabs/tailwindcss#2402] [Help]
Hi Tailwind folks!

Me and my team are experiencing some really annoying performance issues in chrome v85+ dev tools, style tab, when working with local (un-purged) tailwindcss _app_.
Seems like whenever chrome has to parse that large css it just chokes on it for about 30-45s.

I know its not tailwind issue but was wondering if anyone hit this sort of issues during development and maybe we can get to the root of it.

**Comment by ruudk** [comment]
Same issue here on Version 87.0.4280.141 (Official Build) (x86_64)

**Comment by Yori-Mirano** [comment]
I don't know if this will help you, but I just discovered by chance that for me, it was because of activation of the option of Chrome which allows to display a visual representation of the breakpoints of the media queries. See the capture.

It is absolutely necessary to deactivate this option.

![image](https://user-images.githubusercontent.com/1879222/106866052-4ad88f00-66cc-11eb-9ad6-3647760de64a.png)

![image](https://user-images.githubusercontent.com/1879222/106865945-241a5880-66cc-11eb-8d22-f90b3397eda2.png)

**Comment by simonswiss** [comment]
Hey!

I hear you, I was able to reproduce this "Chrome choke" when enabling **all the colors** in the config file (each single color generates a whole lot of utilities, and they are the quickest road to a huge unpurged file.

This is frustrating, while we're contemplating different ways to improve this - the only suggestion I have for now is to reduce the amount of classes you use in your dev environment. Culling colors is where you can do the quickest gains. 

There is a very low probability you're using all for the shades for all of the colors. Unless you're doing a styleguide showing all the available values in the config file, like we're doing on the Tailwind CSS documentation website, which is how we found out about the stretch threshold of Chrome dev tools 😅

I know this is not a great answer, but there is no ideal solutions for this right now. I know @RobinMalfait has been doing some performance spike researches a while back, but with the file size generated with Tailwind 2 + a bunch of extended values, this can become a problem.

Other option is to enable purge at dev time with `purge.enabled = true` in the config, but this will slow another part of your workflow.

Again, if you can cut out a few of the colours, you should see a significant difference in Chrome devtools behaviour.

**Reply by fezproof** [reply]
Is the new JIT mode a solution to this? I have been using it for a few months now and I haven't experienced any of these issues!

**Reply by mgibas** [reply]
Yeah, JIT fixed that by design + introduced couple extra features in the process ;)

**Reply by didac-wh** [reply]
When I use JIT during development, the project doesn't recognize CSS classes until I restart the React server, unless they are used somewhere else before.

**Reply by mgibas** [reply]
How is that related @didac-wh ?

**Reply by simonswiss** [reply]
Yep @fezproof, it was the primary reason for the `JIT` engine to exist, and that solves this problem completely 👍

@didac-wh, you probably need to use the `TAILWIND_MODE=watch` environment variable in the script you run to start your server. If not, please open another discussion after checking the troubleshooting JIT section of the Tailwind docs:

https://tailwindcss.com/docs/just-in-time-mode#troubleshooting
